cli/src/commands/workspace.rs

```diff
@@ -171,6 +171,16 @@ fn cmd_workspace_add(
         "Created workspace in \"{}\"",
         file_util::relative_path(command.cwd(), &destination_path).display()
     )?;
+    // Show a warning if the user passed a path without a separator, since they
+    // may have intended the argument to only be the name for the workspace.
+    if !args.destination.contains(std::path::is_separator) {
+        writeln!(
+            ui.warning_default(),
+            "Workspace created inside current directory. If this was unintentional, delete the \
+             \"{}\" directory and run `jj workspace forget {name}` to remove it.",
+            args.destination
```

Comment:
Nit: Use a raw string here, as simplifies escaping and avoids error-prone newlines.